Fix: Landing on middle of page after clicking a link in footer

banflam opened the pull request:
closes #5991 

Note: I have solved this issue by deleting references to overflow. While this fixed the issue, I don't know why those references were there in the first place and it's possible this approach might introduce some other bug.

I've tested things myself but I've not found any newly-introduced bug.

If someone who wrote that code could comment it would be helpful. Otherwise we can go ahead and merge for now and see if there's any impact later.

**Web frontend checklist**
- [x] Formatted my code with `yarn format`
- [x] There are no warnings from `yarn lint --fix`
- [x] There are no console warnings when running the app
- [x] Clicked around my changes running locally and it works
<!---
Remember to request review from couchers-org/web, couchers-org/backend or an individual.
Once your code is approved, remember to merge it if you have write access
--->

banflam committed:
bug fix: after two days of debugging found that deleting the overflow: hidden style from GlobalStyles and overflowY: auto from PageWrapper fixed the issue of pages accessed by clicking links in the footer going only halfway or not going to the top at all

aapeliv commented:
The navbar now disappears when you scroll down

nabramow commented on app/web/components/AppRoute.tsx:
```diff
@@ -24,7 +24,6 @@ const globalStyles = (
     styles={{
       "html, body": {
         margin: 0,
-        overflow: "hidden", // Prevents whole-page scrolling
```
Styles in this component affect the entire app, so changing here is quite risky. 

This is fixing it by accident, as you are saying "don't allow scrolling in main app page". Of course banning scrolling will fix any weird scrolling, but it will also break scrolling where we should have it.

You're going to want to find the specific component causing the weird scrolling behavior and fix it there instead.

banflam replied:
Thanks for the feedback. I might not be able to look at this bug again until Friday 23rd if that's fine. Please let me know if fixing this bug is part of any release

nabramow commented:
@banflam are you still working on this? No worries if not, I just want to put it back on the board if you're no longer working on it so it doesn't get blocked for too long.

banflam commented:
@nabramow Sorry about the delay. I will work on it this weekend and post an update by Monday midnight EST. Is that alright?

banflam commented:
I don't know how to fix this bug @nabramow . I would be grateful if someone could work through this one with me, so that I can gain a deeper understanding of the codebase and be able to fix bugs like these in the future.

Here are the things that I've tried:

1 - Checked for relevant components having scrollIntoView somewhere, and if those scrollIntoView are causing the issue -- LandingPage.tsx, UserPage.tsx, useMapSearchActions.tsx

2 – Confirmed it was not a simple anchor/hash problem
Footer <Link> elements contain plain paths (/plan, /faq, …) – no #fragment.

3 – Eliminated obvious JavaScript scroll calls
Commented-out every scrollIntoView in Comment.tsx.

Deleted the moreContentRef scroll in LandingPage.

4 – Removed global “wiggle” hack in _app.tsx

window.scrollTo(0, scrollY+1); window.scrollTo(0, scrollY-1);
window.addEventListener("focusout", resetScroll);
Effect: confirmed via console trace that hack was gone, but mid-page landing persisted.

5 – Console-traced all scroll & focus calls
Added monkey-patch for scrollTo / scrollIntoView / focus.

After navigation the only call is Next.js’ own handleSmoothScroll → scrollTo(0,0).

No further scroll/focus calls occur; therefore JS is not pushing us down.

6 – Forced top scroll after route change (several variants)

router.events.on("routeChangeComplete", () => window.scrollTo(0,0))

Same with setTimeout(...,150) and with scrollRestoration="manual"

7 – Ruled out scroll={false}
Added scroll={true} to our StyledLink wrapper so every internal link explicitly requests top scroll – still jumps.

8 – Disabled viewport-height CSS hack
Commented-out updateVH() effect that rewrites --vh (mobile 100 vh workaround) → no change.

Observation: each route always lands at the same fixed offset (e.g., page A ≈75 %, page B bottom).

nabramow commented:
Closing as handled in #6216
